src/middlewares/cart.middleware.js

```diff
@@ -0,0 +1,62 @@
+import { CartService } from '../services/cart.service';
+import { ProductService } from '../services/product.service';
+
+export const getUserCart = async (req, res, next) => {
+	const { id: userId } = req.user;
+	let { productId, productQuantity } = req.body;
+	if (!productQuantity) {
+		productQuantity = 1;
+	}
+	let userCart = await CartService.getCart(userId);
+	if (!userCart) {
+		const total = 0;
+		const products = [{ quantity: productQuantity, productId: productId }];
+		const cart = {
+			userId,
+			total,
+			products,
+		};
+		userCart = await CartService.createCart(cart);
+	}
+	req.cart = userCart;
+	next();
+};
```

Comment:
This middleware should only be used to get the user's cart. 
Hint: And if there is no cart, return next()